src/Stack/Upgrade.hs

```diff
@@ -42,7 +45,16 @@ upgrade gitRepo mresolver = withCanonicalizedSystemTempDirectory "stack-upgrade"
       Just repo -> do
         remote <- liftIO $ readProcess "git" ["ls-remote", repo, "master"] []
         let latestCommit = head . words $ remote
-        if latestCommit == $gitHash then do
+#ifdef USE_GIT_INFO
+            builtHash = $gitHash
+#else
+            builtHash = "git information disabled by flag"
```

Comment:
Maybe this warning should also be displayed for the case when no git version information is available (e.g. if you built from hackage directly or from an sdist).